python/ml4ir/applications/classification/tests/test_classification_model.py

```diff
@@ -8,9 +8,9 @@ class ClassificationModelTest(ClassificationTestBase):
     :func:`~ml4ir.applications.classification.pipeline.ClassificationPipeline.get_relevance_model`
     """
 
-    def test_csv_loss(self):
+    def test_csv_loss_metrics(self):
```

Comment:
I followed Arvind's suggestion here and made the 3 functions one, to avoid executing setUp thrice. It speeds up test ~40 sec.

Reply:
We should use [subtest](https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest) here. It'll run all tests in parallel and report it individually

Reply:
thanks for the hint, I was not aware of it though. It will have to be part of another PR though, I have alredy spent much time on this one.